Mage.Sets/src/mage/cards/l/LiberatedLivestock.java

```diff
@@ -0,0 +1,121 @@
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{5}{W}");
+        
+        this.subtype.add(SubType.CAT);
+        this.subtype.add(SubType.BIRD);
+        this.subtype.add(SubType.OX);
+        this.power = new MageInt(4);
+        this.toughness = new MageInt(6);
+
+
+
+        // When Liberated Livestock dies, create a 1/1 white Cat creature token with lifelink,
+        // a 1/1 white Bird creature token with flying,
+        // and a 2/4 white Ox creature token.
+        // For each of those tokens, you may put an Aura card from your hand and/or graveyard onto the battlefield attached to it.
+        this.addAbility(new DiesSourceTriggeredAbility(new LiberatedLivestockEffect()));
+    }
+
+    private LiberatedLivestock(final LiberatedLivestock card) {
+        super(card);
+    }
+
+    @Override
+    public LiberatedLivestock copy() {
+        return new LiberatedLivestock(this);
+    }
+}
+
+class LiberatedLivestockEffect extends OneShotEffect {
+
+    LiberatedLivestockEffect() {
+        super(Outcome.PutCreatureInPlay);
+        staticText = "create a 1/1 white Cat creature token with lifelink, a 1/1 white Bird creature token with flying, and a 2/4 white Ox creature token. For each of those tokens, you may put an Aura card from your hand and/or graveyard onto the battlefield attached to it.";
+    }
+
+    private LiberatedLivestockEffect(final LiberatedLivestockEffect effect) {super(effect);}
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer((source.getControllerId()));
+        if (controller == null){
+            return false;
+        }
+        Token cat = new CatToken2();
+        Token bird = new BirdToken();
+        Token ox = new OxToken();
+
+        // pause to get token onto the battlefield all the same time
+        game.getState().pause();
+        cat.putOntoBattlefield(1, game, source, source.getControllerId());
+        bird.putOntoBattlefield(1, game, source, source.getControllerId());
+        ox.putOntoBattlefield(1, game, source, source.getControllerId());
+        game.getState().resume();
+
+        // Setup for choosing and attaching auras to tokens
+        List<Token> tokens = Arrays.asList(cat, bird, ox);
+        FilterCard filter = new FilterCard("Up to one Aura");
+        filter.add(SubType.AURA.getPredicate());
+        TargetCard target;
+
+
+        for (Token token : tokens){
+            for (UUID t : token.getLastAddedTokenIds()){
+                Permanent sourcePermanent = game.getPermanent(t);
+                if (sourcePermanent == null){continue;}
+                filter.setMessage("Aura from you hand or graveyard that can attach to " + sourcePermanent.getName());
+
+                // Choose from hand or graveyard?
+                if (controller.chooseUse(outcome, "Look in Hand or Graveyard for aura to attach to "+sourcePermanent.getName()+"?", null, "Hand", "Graveyard", source, game)){
```

Comment:
Clunky workaround, but it's okay given engine limitations

Reply:
I tried to implement a way to target from two different zones, but it took too long, and I don't know if I even would be able.
As far as I know this implementation is the standard I looked at the implementation of around 12 other cards that look in hand or graveyard 
(The list of cards if usefull) https://scryfall.com/search?q=o%3A%2Ffrom+your+hand.*or.*graveyard%2F&unique=cards&as=grid&order=name

Reply:
I also agree that asking for the zone is easier for now.
In the future we might enhance multizone selection overall.

Reply:
Hm, actually, what about something like:
```java
Set<Card> cards = controller.getHand().getCards(filter, game);
cards.addAll(controller.getGraveyard().getCards(filter, game));
TargetCard target = new TargetCard(0, 1, Zone.ALL, filter);
target.withNotTarget(true);
target.withChooseHint("to attach to " + sourcePermanent.getName());
controller.choose(outcome, cards, target, source, game);
Card card = cards.get(target.getFirstTarget(), game);
```

(and can make the filter static here)